File: learning-javascript/leet-code/src/is-subsequence/is-subsequence.ts
function isSubsequence(s: string, t: string): boolean {
  if (s === t) {
    return true;
  }

  return recurse(s, t);
}

function recurse(remainingSubsequence: string, remainingString: string) {
  if (remainingSubsequence.length > remainingString.length) {
    return false;
  }
  if (remainingSubsequence.length === 0) {
    return true;
  }

  for (let i = 0; i < remainingString.length; i++) {
    if (remainingSubsequence[0] === remainingString[i]) {
      return recurse(remainingSubsequence.slice(1), remainingString.slice(i + 1));
    }
  }
  return false;
}

File: katas/isSubsequence/isSubsequence.py
def isSubsequence(s, t):
  if s == t or s =='':
    return True

  return recurse(s, t)

def recurse(remainingSubsequence, remainingString):
  if len(remainingSubsequence) > len(remainingString):
    return False
 
  if len(remainingSubsequence) == 0:
    return True

  for i in range(len(remainingString)):
    if remainingSubsequence[0] == remainingString[i]:
      return recurse(remainingSubsequence[1:], remainingString[i+1:])
  return False